File: src/steps/store-splitted-abc-converted-orig.ts
import type { TaraskStep, SplittedTextStorage } from './types';

export const storeSplittedAbcConvertedOrig: TaraskStep<SplittedTextStorage> = ({
	cfg: {
		abc: { lower, upper },
	},
	text,
	storage,
}) => {
	text = lower(text);
	storage.origArr = (upper ? upper(text) : text).split(' ');
};

File: taraskevizer-rs/crates/core/src/pipeline/steps/store_splitted_abc_converted_orig.rs
use crate::pipeline::{
    helpers::{alphabet_dict, apply_alphabet},
    PipelineContext,
};

pub fn step_store_splitted_abc_converted_orig(ctx: &mut PipelineContext) {
    let lowered = apply_alphabet(&ctx.text, ctx.cfg.abc, "lower");
    let upper_dict = alphabet_dict(ctx.cfg.abc, "upper");
    let converted = if upper_dict.has_entries() {
        upper_dict.replace_all(&lowered)
    } else {
        lowered
    };
    ctx.orig_arr = converted.split(' ').map(|s| s.to_string()).collect();
}
